tribe-ext-show-cost-field.php

```diff
@@ -0,0 +1,137 @@
+ * License URI:       https://www.gnu.org/licenses/gpl-3.0.html
+ * Text Domain:       tribe-ext-show-cost-field
+ *
+ *     This plugin is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     any later version.
+ *
+ *     This plugin is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *     GNU General Public License for more details.
+ */
+
+// Do not load unless Tribe Common is fully loaded and our class does not yet exist.
+if (
+	class_exists( 'Tribe__Extension' )
+	&& ! class_exists( 'Tribe__Extension__Show_Cost_Field' )
+) {
+	/**
+	 * Extension main class, class begins loading on init() function.
+	 */
+	class Tribe__Extension__Show_Cost_Field extends Tribe__Extension {
+
+		/**
+		 * Setup the Extension's properties.
+		 *
+		 * This always executes even if the required plugins are not present.
+		 */
+		public function construct() {
+			$this->set_url( 'https://theeventscalendar.com/extensions/show-cost-field/' );
+			add_action( 'tribe_plugins_loaded', array( $this, 'required_tribe_classes' ), 0 );
+		}
+
+		/**
+		 * Check required plugins after all Tribe plugins have loaded.
+		 */
+		public function required_tribe_classes() {
+			// If neither Event Tickets or Eventbrite Tickets, require Event Tickets
+			if (
+				! Tribe__Dependency::instance()->is_plugin_active( 'Tribe__Tickets__Main' )
+				&& ! Tribe__Dependency::instance()->is_plugin_active( 'Tribe__Events__Tickets__Eventbrite__Main' )
+			) {
+				add_filter( 'gettext_with_context', array( $this, 'reqd_plugins_and_to_or_text' ), 10, 4 );
+				$this->add_required_plugin( 'Tribe__Tickets__Main' );
+				$this->add_required_plugin( 'Tribe__Events__Tickets__Eventbrite__Main' );
+			}
+
+			// Event Tickets
+			if ( Tribe__Dependency::instance()->is_plugin_active( 'Tribe__Tickets__Main' ) ) {
+				$this->add_required_plugin( 'Tribe__Tickets__Main' );
+			}
+
+			// Eventbrite Tickets
+			if ( Tribe__Dependency::instance()->is_plugin_active( 'Tribe__Events__Tickets__Eventbrite__Main' ) ) {
+				$this->add_required_plugin( 'Tribe__Events__Tickets__Eventbrite__Main' );
+			}
+		}
+
+		/**
+		 * We require only one of the required plugins so we change the
+		 * conjunction from "and" to "or".
+		 *
+		 * @see Tribe__Admin__Notice__Plugin_Download::implode_with_grammar()
+		 *
+		 * @param $translation
+		 * @param $text
+		 * @param $context
+		 * @param $domain
+		 *
+		 * @return string
+		 */
+		public function reqd_plugins_and_to_or_text( $translation, $text, $context, $domain ) {
+			if (
+				'tribe-common' == $domain
+				&& ' and ' == $translation
+				&& 'the final separator in a list of two or more items' == $context
+			) {
+				$translation = ' or '; // OR, not AND
```

Comment:
* If another extension is also active and requires multiple plugins, won't this also impact that other message?
* I'm guessing we're using English only (we're not testing for translated forms of _'and'_ or _'or')_ because this isn't i18n-safe for all languages ... but that in itself could be an indicator that this is not the ideal approach

Reply:
1. Yes, I thought of this but we haven't done this in the past and it's probably fairly rare. I tried to unhook but then it didn't take effect at all. Any ideas how to unhook to not affect others using this?
2. Good catch! Would changing the priority from `10` to `0` address this?

Reply:
It _looks_ like `remove_filter( 'gettext_with_context', array( $this, 'reqd_plugins_and_to_or_text' ) )` would be sufficient. If that's not working and in any case conflicts with similar messages from other extensions are rare, we can just skip this.

Reply:
that's what I tried. it didn't cause problems, just removed it before it changed it I guess.